omnibus/config/software/haproxy.rb

```diff
@@ -44,29 +44,19 @@
    "USE_PCRE_JIT" => "1",
    "USE_ZLIB" => "1",
    "USE_OPENSSL" => "1",
+   # Required to resolve hostnames to IPv6 addresses
+   # off-by-default because of prolems on older glibc's
+   "USE_GETADDRINFO" => "1",
+   "TARGET" => "linux2628"
   }
-  # Required to resolve hostnames to IPv6 addresses
-  # off-by-default because of prolems on older glibc's
-  # TODO(ssd): Should we turn this off on RHEL5?
-  build_options['USE_GETADDRINFO'] = "1"
+
   if intel?
     build_options["USE_REGPARM"] = "1"
   end
-  build_options['TARGET'] = if ohai["kernel"] && ohai["kernel"]["name"] == "Linux"
-                              version = Gem::Version.new(String(ohai["kernel"]["release"]).split("-").first)
-                              case
-                              when version >= Gem::Version.new("2.6.28")
-                                "linux2628"
-                              when version >= Gem::Version.new("2.6")
-                                "linux26"
-                              else
-                                "linux24e"
-                              end
-                            else
-                              "generic"
-                            end
+
   build_args = ""
   build_options.each { |k,v| build_args << " #{k}=#{v}"}
   make "haproxy #{build_args}", env: env
+  make "#{build_args}", env: env
```

Comment:
What's the intent here?  I think the default `all` target in the HAProxy Makefile is equivalent to the `haproxy` target, so it seems to me this would be building haproxy twice.

Reply:
Builds fail when haproxy got bumped to `1.6.15` with the following error:

```
The following shell command exited with status 2:
 
    $ CFLAGS=-I/opt/opscode/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector CPPFLAGS=-I/opt/opscode/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector CXXFLAGS=-I/opt/opscode/embedded/include -O2 -D_FORTIFY_SOURCE=2 -fstack-protector LDFLAGS=-Wl,-rpath,/opt/opscode/embedded/lib -L/opt/opscode/embedded/lib LD_RUN_PATH=/opt/opscode/embedded/lib MAKE=gmake OMNIBUS_INSTALL_DIR=/opt/opscode PATH=/opt/opscode/bin:/opt/opscode/embedded/bin:/var/lib/buildkite-agent/builds/buildkite-omnibus-rhel-7-x86-64-i-0ba65d07907deb572-1/chef/chef-chef-server-master-omnibus-adhoc/omnibus/vendor/bundle/ruby/2.6.0/bin:/opt/omnibus-toolchain/bin:/usr/bin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin PKG_CONFIG_PATH=/opt/opscode/embedded/lib/pkgconfig gmake install  PREFIX=/opt/opscode/embedded USE_PCRE=1 USE_PCRE_JIT=1 USE_ZLIB=1 USE_OPENSSL=1 USE_GETADDRINFO=1 TARGET=linux2628 USE_REGPARM=1
 
Output:
 
    Please run 'make' before 'make install'.

```

and can be seen in this failed build:

https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/961

This seems to be related to this patch (thanks @PrajaktaPurohit!):

http://git.haproxy.org/?p=haproxy-1.6.git;a=commitdiff;h=6f54ee4

Reply:
In that case, can we remove the previous make line?

Reply:
Quite likely and I'll verify in my next adhoc build after rebasing to include a fix currently being worked on